picard/tagger.py

```diff
@@ -1086,29 +1099,42 @@ def main(localedir=None, autoupdate=True):
     signal.signal(signal.SIGINT, signal.SIG_DFL)
 
     picard_args, unparsed_args = process_picard_args()
+
     if picard_args.version:
         return version()
     if picard_args.long_version:
         return longversion()
 
-    try:
-        from PyQt5.QtDBus import QDBusConnection
-        dbus = QDBusConnection.sessionBus()
-        dbus.registerService(PICARD_APP_ID)
-    except ImportError:
-        pass
-
-    tagger = Tagger(picard_args, unparsed_args, localedir, autoupdate)
+    should_start = picard_args.stand_alone_instance
 
-    # Initialize Qt default translations
-    translator = QtCore.QTranslator()
-    locale = QtCore.QLocale()
-    translation_path = QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryLocation.TranslationsPath)
-    log.debug("Looking for Qt locale %s in %s", locale.name(), translation_path)
-    if translator.load(locale, "qtbase_", directory=translation_path):
-        tagger.installTranslator(translator)
+    if not should_start:
+        pipe_handler = pipe.Pipe(app_name=PICARD_APP_NAME, app_version=PICARD_FANCY_VERSION_STR, args=picard_args.FILE)
+        should_start = True in [pipe_handler.permission_error_happened, pipe_handler.is_pipe_owner]
+        if pipe_handler.permission_error_happened:
+            pipe_handler = None
     else:
-        log.debug('Qt locale %s not available', locale.name())
+        pipe_handler = None
+
+    # No `else` statement is needed since pipe.Pipe has already sent picard_args to the existing instance
+    if should_start:
+        try:
+            from PyQt5.QtDBus import QDBusConnection
+            dbus = QDBusConnection.sessionBus()
+            dbus.registerService(PICARD_APP_ID)
+        except ImportError:
+            pass
+
+        tagger = Tagger(picard_args, unparsed_args, localedir, autoupdate, pipe_handler=pipe_handler)
+
+        # Initialize Qt default translations
+        translator = QtCore.QTranslator()
+        locale = QtCore.QLocale()
+        translation_path = QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryLocation.TranslationsPath)
+        log.debug("Looking for Qt locale %s in %s", locale.name(), translation_path)
+        if translator.load(locale, "qtbase_", directory=translation_path):
+            tagger.installTranslator(translator)
+        else:
+            log.debug('Qt locale %s not available', locale.name())
 
-    tagger.startTimer(1000)
-    sys.exit(tagger.run())
+        tagger.startTimer(1000)
+        sys.exit(tagger.run())
```

Comment:
Okay, works for me, there are 2 options:

1) `sys.exit(0)` because it executed correctly.
2) some custom free-to-use error code to let user know wheter there was already an instance running. 

2nd one is better imo, can someone think of an example of such code? I tried finding it out myself but [PyQT5.QtWidgets.QApplication docs](https://www.riverbankcomputing.com/static/Docs/PyQt5/api/qtwidgets/qapplication.html#) are marked TODO...

Reply:
Made it 30403, shouldn't collide with anything else.